MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java

```diff
@@ -0,0 +1,290 @@
+package com.mapbox.mapboxandroiddemo.examples.labs;
+
+import android.animation.TimeInterpolator;
+import android.animation.ValueAnimator;
+import android.graphics.BitmapFactory;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.view.animation.AccelerateInterpolator;
+import android.view.animation.BounceInterpolator;
+import android.view.animation.DecelerateInterpolator;
+import android.view.animation.LinearInterpolator;
+
+import com.getbase.floatingactionbutton.FloatingActionButton;
+import com.mapbox.geojson.Feature;
+import com.mapbox.geojson.FeatureCollection;
+import com.mapbox.geojson.Point;
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconAllowOverlap;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconIgnorePlacement;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconImage;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconOffset;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconTranslate;
+
+/**
+ * Combine SymbolLayer icons with the Android system's ValueAnimator and interpolator
+ * animation for a fun pin drop effect. The interpolator movement can also be used with other
+ * types of map layers, such as a LineLayer or CircleLayer.
+ * <p>
+ * More info about https://developer.android.com/reference/android/view/animation/Interpolator
+ */
+public class ValueAnimatorIconAnimationActivity extends AppCompatActivity implements
+  OnMapReadyCallback, MapView.OnDidFinishRenderingMapListener {
+
+  private static final String ICON_ID = "red-pin-icon-id";
+
+  // This float's actual value will depend on the height of the SymbolLayer icon
+  private static final float DEFAULT_DESIRED_ICON_OFFSET = -16;
+  private static final float STARTING_DROP_HEIGHT = -100;
+  private static final long DROP_SPEED_MILLISECONDS = 1200;
+  private MapView mapView;
+  private SymbolLayer pinSymbolLayer;
+  private Style style;
+  private TimeInterpolator currentSelectedTimeInterpolator;
+  private ValueAnimator animator;
+  private boolean firstRunThrough;
+  private boolean animationHasStarted;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    currentSelectedTimeInterpolator = new BounceInterpolator();
```

Comment:
can be assigned inline in class member definition

Reply:
Fixed